### PR TITLE
Disney+: big update

### PR DESCRIPTION
**Pull request by ThaTiemsz**
Big update for the Disney+ presence including:

* Refactoring
* Added settings
* Buttons!
* Support for the Disney+ GroupWatch feature
* New thumbnail and lots more descriptions in different languages

New button
GroupWatch lobby and buttons
GroupWatching and buttons
Settings

### CODE REVIEW

File: websites/D/Disney+/dist/metadata.json [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
+    "no": "Disney+ er det eksklusive hjemmet for favorittfilmene og seriene dine fra Disney, Pixar, Marvel, Star Wars, National Geographic og Star. Begynn å strømme i dag.",
+    "fi": "Disney+ tarjoaa suosikkielokuviasi ja tv-ohjelmia yksinoikeudella tekijöiltä Disney, Pixar, Marvel, Star Wars, National Geographic ja Star. Aloita katselu tänään.",
+    "sv_SE": "Disney+ är det enda hemmet för dina favoritfilmer och tv-program från Disney, Pixar, Marvel, Star Wars, National Geographic och Star. Börja streama idag.",
+    "da_DK": "Disney+ er det eksklusive hjem for dine yndlingsfilm og -tv-serier fra Disney, Pixar, Marvel, Star Wars, National Geographic og Star. Begynd at se i dag.",
+    "pt_BR": "O Disney+ é o serviço exclusivo para ver os seus filmes e programas preferidos da Disney, Pixar, Marvel, Star Wars e National Geographic. Comece a assistir hoje mesmo."
   },
   "url": "www.disneyplus.com",
   "regExp": "([a-z0-9-]+[.])*disneyplus[.]com[/]|([a-z0-9-]+[.])*hotstar[.]com[/]",
-  "version": "1.3.7",
+  "version": "1.3.8",
```
Comment (Bas950) [inline websites/D/Disney+/dist/metadata.json:24]: 2.0.0
or 1.4.0

File: websites/D/Disney+/presence.ts [hunk 2]
```diff
@@ -2,33 +2,51 @@ const presence: Presence = new Presence({
...
+    },
+    await presence.getSetting("lang")
+  );
 }
 
+let strings: LangStrings;
+let oldLang: string;
 let title: string;
 let subtitle: string;
+let groupWatchCount: number;
```
Comment (Bas950) [inline websites/D/Disney+/presence.ts:30]: merge lets

File: websites/D/Disney+/presence.ts [hunk 3]
```diff
@@ -2,33 +2,51 @@ const presence: Presence = new Presence({
...
 
 presence.on("UpdateData", async () => {
+  const newLang: string = await presence.getSetting("lang");
+  const time: boolean = await presence.getSetting("time");
+  const buttons: boolean = await presence.getSetting("buttons");
+  const groupWatchBtn: boolean = await presence.getSetting("groupWatchBtn");
   const isHostDP = /(www\.)?disneyplus\.com/.test(location.hostname);
   const isHostHS = /(www\.)?hotstar\.com/.test(location.hostname);
-  const data: PresenceData = {};
+  const data: PresenceData & {
```
Comment (Bas950) [inline websites/D/Disney+/presence.ts:39]: merge consts

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/D/Disney+/dist/metadata.json [hunk 4]
```diff
@@ -1,22 +1,29 @@
...
+    "pt": "O Disney+ é a morada exclusiva das suas séries e filmes preferidos da Disney, Pixar, Marvel, Star Wars, National Geographic e Star. Comece já a ver.",
+    "no": "Disney+ er det eksklusive hjemmet for favorittfilmene og seriene dine fra Disney, Pixar, Marvel, Star Wars, National Geographic og Star. Begynn å strømme i dag.",
+    "fi": "Disney+ tarjoaa suosikkielokuviasi ja tv-ohjelmia yksinoikeudella tekijöiltä Disney, Pixar, Marvel, Star Wars, National Geographic ja Star. Aloita katselu tänään.",
+    "sv_SE": "Disney+ är det enda hemmet för dina favoritfilmer och tv-program från Disney, Pixar, Marvel, Star Wars, National Geographic och Star. Börja streama idag.",
+    "da_DK": "Disney+ er det eksklusive hjem for dine yndlingsfilm og -tv-serier fra Disney, Pixar, Marvel, Star Wars, National Geographic og Star. Begynd at se i dag.",
+    "pt_BR": "O Disney+ é o serviço exclusivo para ver os seus filmes e programas preferidos da Disney, Pixar, Marvel, Star Wars e National Geographic. Comece a assistir hoje mesmo."
```
Comment (ririxidev) [inline websites/D/Disney+/dist/metadata.json:20]: you know all of these languages or you just used translator? Remove them if I'm right

File: websites/D/Disney+/presence.ts [hunk 5]
```diff
@@ -37,85 +55,149 @@ presence.on("UpdateData", async () => {
...
       data.details = title;
-      data.state = subtitle ? subtitle : "Movie";
-
+      data.state = subtitle || "Movie";
       data.smallImageKey = video.paused ? "pause" : "play";
       data.smallImageText = video.paused
-        ? (await strings).pause
-        : (await strings).play;
+        ? strings.pause
+        : strings.play;
```
Comment (ririxidev) [inline websites/D/Disney+/presence.ts:179]: do await for sure...

File: websites/D/Disney+/presence.ts [hunk 6]
```diff
@@ -37,85 +55,149 @@ presence.on("UpdateData", async () => {
...
+        }];
+      }
+
       if (title) presence.setActivity(data, !video.paused);
     }
+
+  // Browsing
   } else {
-    data.details = (await strings).browsing;
+    data.details = strings.browsing;
```
Comment (ririxidev) [inline websites/D/Disney+/presence.ts:200]: do await for sure...

File: websites/D/Disney+/presence.ts [hunk 7]
```diff
@@ -37,85 +55,149 @@ presence.on("UpdateData", async () => {
...
+      } else {
+        data.details = title;
+        data.state = subtitle || "Movie";
+      }
 
       data.smallImageKey = video.paused ? "pause" : "play";
-      data.smallImageText = video.paused
-        ? (await strings).pause
-        : (await strings).play;
+      data.smallImageText = video.paused ? strings.pause : strings.play;
```
Comment (ririxidev) [inline websites/D/Disney+/presence.ts:88]: do await for sure...

File: websites/D/Disney+/presence.ts [hunk 8]
```diff
@@ -2,33 +2,51 @@ const presence: Presence = new Presence({
...
-  const isHostHS = /(www\.)?hotstar\.com/.test(location.hostname);
-  const data: PresenceData = {};
+  const newLang: string = await presence.getSetting("lang"),
+        time: boolean = await presence.getSetting("time"),
+        buttons: boolean = await presence.getSetting("buttons"),
+        groupWatchBtn: boolean = await presence.getSetting("groupWatchBtn"),
+        isHostDP = /(www\.)?disneyplus\.com/.test(location.hostname),
+        isHostHS = /(www\.)?hotstar\.com/.test(location.hostname),
+        data: PresenceData & {
+          partyId?: string;
```
Comment (ririxidev) [inline websites/D/Disney+/presence.ts:40]: you don't use ``partyId`` anywhere...

**Review by ririxidev — CHANGES_REQUESTED** [review]
You should look at these things:

File: websites/D/Disney+/dist/metadata.json (hunk 4 above)
Comment (ThaTiemsz) [reply websites/D/Disney+/dist/metadata.json:20]: These come straight from the Disney+ site (minus the mention of "Star" that I added). I changed my profile's language for each of the languages to get the strings from the source code.

File: websites/D/Disney+/presence.ts (hunk 5 above)
Comment (ThaTiemsz) [reply websites/D/Disney+/presence.ts:179]: `await` is already used on L48 and therefore not needed any further, but I could change it back if that's preferred.

File: websites/D/Disney+/presence.ts (hunk 8 above)
Comment (ThaTiemsz) [reply websites/D/Disney+/presence.ts:40]: Forgot to remove this, thanks

**Review by Bas950 — APPROVED** [review]

**Review by ririxidev — APPROVED** [review]

**Comment by monish-1234** [comment]
Presence not working after update in www.hotstar.com

**Comment by ThaTiemsz** [comment]
> Presence not working after update in [www.hotstar.com](http://www.hotstar.com)

Thanks, I'll investigate.

**Pull request merged**